### PR TITLE
[TOPI] Fix nn.lrn result dtype on fp16

### PR DESCRIPTION
**Pull request by wrongtest**
The buggy script as below:
```python
import tvm
from tvm import relay
from tvm.contrib import graph_executor
x = relay.var("x", shape=[1, 3, 224, 224], dtype="float16")
y = relay.nn.lrn(x)
mod = tvm.IRModule.from_expr(relay.Function([x], y))
lib = relay.build(mod, target="llvm")
f = graph_executor.GraphModule(lib["default"](tvm.cpu()))
f.run()
```

The error I get is
```
Check failed: ret == 0 (-1 vs. 0) : Assert fail: (((tir.tvm_struct_get(arg.T_divide, 0, 5) == (uint8)2) && (tir.tvm_struct_get(arg.T_divide, 0, 6) == (uint8)32)) && (tir.tvm_struct_get(arg.T_divide, 0, 7) == (uint16)1)), arg.T_divide.dtype is expected to be float32
```

### CODE REVIEW

File: tests/python/topi/python/test_topi_lrn.py [hunk 1]
```diff
@@ -66,6 +65,7 @@ def test_lrn():
     verify_lrn((1, 3, 5, 5), 3, 1, 1.0, 1.0, 0.5)
     verify_lrn((1, 3, 5, 5), 3, 3, 1.0, 1.0, 0.5)
     verify_lrn((1, 3, 20, 20), 3, 1, 2.0, 1.0, 0.75)
+    verify_lrn((1, 3, 5, 5), 3, 3, 1.0, 1.0, 0.5, dtype="float16", rtol=1e-3, atol=1e-3)
```
Comment (MasterJH5574) [inline tests/python/topi/python/test_topi_lrn.py:68]: Do you mean the 1e-3 tolerance doesn’t work for this set of parameters?

**Review by MasterJH5574 — COMMENTED** [review]
Thanks for the fix! The code example you provided can run successfully on my machine. Looks like you should use `float16` to reproduce the error :eyes:.

Besides that I have one minor question. Would be great if you can have a look, thanks!

File: tests/python/topi/python/test_topi_lrn.py (hunk 1 above)
Comment (wrongtest) [reply tests/python/topi/python/test_topi_lrn.py:68]: Yes, It seems that original rtol 1e5 not work with f16. It failed on my machine like
```
Mismatched elements: 4 / 75 (5.33%)
Max absolute difference: 0.000977
Max relative difference: 0.001402
 x: array([[[[2.761e-01, 3.909e-01, 2.979e-01, 1.460e-01, 1.453e-01],
         [7.842e-01, 7.104e-01, 3.000e-01, 7.490e-01, 5.640e-01],
         [1.940e-01, 1.163e-01, 6.709e-01, 1.050e-01, 5.229e-01],...
 y: array([[[[2.761e-01, 3.909e-01, 2.979e-01, 1.460e-01, 1.453e-01],
         [7.842e-01, 7.100e-01, 3.000e-01, 7.490e-01, 5.640e-01],
         [1.940e-01, 1.163e-01, 6.709e-01, 1.050e-01, 5.229e-01],...
```

Comment (MasterJH5574) [reply tests/python/topi/python/test_topi_lrn.py:68]: Got it. Thanks for the explanation!

**Review by MasterJH5574 — APPROVED** [review]
LGTM.

**Pull request merged**